packages/core-bridge/src/conversions.rs

```diff
@@ -533,4 +585,43 @@ impl ObjectHandleConversionsExt for Handle<'_, JsObject> {
             )),
         }
     }
+
+    fn as_slot_supplier(&self, cx: &mut FunctionContext) -> NeonResult<SlotSupplierOptions> {
+        if let Some(num_slots) = js_optional_value_getter!(cx, self, "numSlots", JsNumber) {
```

Comment:
As a general rule, I'd prefer to avoid duck-typing the config, and even more so from the Rust side. Let's add a `type` discriminator property in `SlotSupplier`, similar to what we have in `EphemeralServerConfig`. I'd even be inclined to expose that property to users (i.e. they would have to specify the `type` property when explicitly passing a `WorkerOptions.tunner`).

Reply:
Hmm, I did have this earlier and then I stopped doing it specifically because forcing the user to type it seemed a little annoying. If we're fine with that, then I'll add it back.

Reply:
I'm ok with making `type` optional on the user facing API. However, I'd definitely want to avoid doing duck-typing in bridge. That's not that bad, really, but I strive at making the TS/Rust interface easier to maintain/less error-prone, and I feel that having Rust code guess the type of objects based on the presence or not of some property kind goes against that goal.